pkg/initializers/dataset/cache_initalizer.py

```diff
@@ -0,0 +1,209 @@
+import logging
+import os
+from typing import Optional
+
+import yaml
+import time
+from kubernetes import client, config, utils as k8s_utils
+from kubernetes.client.rest import ApiException
+from kubernetes.utils import FailToCreateError
+
+import pkg.initializers.types.types as types
+import pkg.initializers.utils.utils as utils
+
+logging.basicConfig(
+    format="%(asctime)s %(levelname)-8s [%(filename)s:%(lineno)d] %(message)s",
+    datefmt="%Y-%m-%dT%H:%M:%SZ",
+    level=logging.INFO,
+)
+
+
+class CacheInitializer(utils.DatasetProvider):
+
+    def load_config(self):
+        config_dict = utils.get_config_from_env(types.CacheDatasetInitializer)
+        self.config = types.CacheDatasetInitializer(**config_dict)
+
+    def download_dataset(self):
+        logging.info(f"Cache initializer called with storage URI: {self.config.storage_uri}")
+
+        # Check required fields
+        if not self.config.cache_image:
+            logging.error("CACHE_IMAGE environment variable is required but not provided")
+            raise ValueError("CACHE_IMAGE environment variable is required")
+        
+        if not self.config.iam_role:
+            logging.error("IAM_ROLE environment variable is required but not provided")
+            raise ValueError("IAM_ROLE environment variable is required")
+
+        train_job_name = self.config.train_job_name or "cache-test"
+        cache_image = self.config.cache_image
+        cluster_size = self.config.cluster_size or "3"
+        iam_role = self.config.iam_role
+        head_cpu = self.config.head_cpu or "1"
+        head_mem = self.config.head_mem or "1Gi"
+        worker_cpu = self.config.worker_cpu or "2"
+        worker_mem = self.config.worker_mem or "2Gi"
+        
+        substitutions = {
+            'NAME': train_job_name,
+            'IAM_ROLE': iam_role,
+            'SIZE': cluster_size,
+            'IMAGE': cache_image,
+            'HEAD_CPU': head_cpu,
+            'HEAD_MEM': head_mem,
+            'WORKER_CPU': worker_cpu,
+            'WORKER_MEM': worker_mem,
+        }
+        
+        if self.config.metadata_loc:
+            substitutions['METADATA_LOC'] = self.config.metadata_loc
+        if self.config.table_name:
+            substitutions['TABLE_NAME'] = self.config.table_name
+        if self.config.schema_name:
+            substitutions['SCHEMA_NAME'] = self.config.schema_name
+
+        deploy_lws_with_substitution(
+            train_job_name,
+            'cache-initializer-template.yaml',
+            namespace='cache-test',
```

Comment:
namespace must be equal to the TrainJob's namespace.

Reply:
Updated to fetch from env var. Downward api can be used similar to train job name